guide/arabic/certifications/front-end-libraries/react-and-redux/extract-state-logic-to-redux/index.md

```diff
@@ -10,26 +10,26 @@ localeTitle: استخراج الدولة المنطق إلى Redux
 
 الحل المقترح:
 
- `const ADD = 'ADD'; 
- 
- function addMessage(message) { 
-  return { 
-    type: ADD, 
-    message: message 
-  }; 
- }; 
- 
- function messageReducer (previousState, action) { 
-  return [...previousState, action.message]; 
- } 
- 
- let store = { 
-  state: [], 
-  getState: () => store.state, 
-  dispatch: (action) => { 
-    if (action.type === ADD) { 
-      store.state = messageReducer(store.state, action); 
-    } 
-  } 
- }; 
-`
\ No newline at end of file
+ `const ADD = 'ADD';
+
+ const addMessage = (message) => {
+   return {
+     type: ADD,
+     message
+   }
+ };
+
+ const messageReducer = (state = [], action) => {
+   switch (action.type) {
+     case ADD:
+       return [
+         ...state,
+         action.message
+       ];
+     default:
+       return state;
+   }
+ };
+
+ const store = Redux.createStore(messageReducer); 
+`
```

Comment:
Make this three backticks

Reply:
I realized this, but I noticed the other guides I looked at in these languages didn't use the three backticks, so I left it like this for consistency and figured there might've been a reason for it.

Reply:
Nope, this is fine. Please change it so it reflects the language too